### PR TITLE
switch to pathlib and add Pytests aimec

### PR DESCRIPTION
**Pull request by matthiasto**
switched from os.path.join to pathlib.Path.: in aimec, com2AB, in2Trans 
Added a pytest for read aimec inputs

### CODE REVIEW

File: avaframe/com1DFA/com1DFA.py [hunk 1]
```diff
@@ -1810,10 +1810,10 @@ def readPartFromPickle(inDir, flagAvaDir=False):
     """
 
     if flagAvaDir:
-        inDir = os.path.join(inDir, 'Outputs', 'com1DFA', 'particles')
+        inDir = pathlib.Path(inDir, 'Outputs', 'com1DFA', 'particles')
 
     # search for all pickles within directory
-    PartDicts = sorted(glob.glob(os.path.join(inDir, '*.p')))
+    PartDicts = sorted(inDir.glob('*.p'))
```
Comment (awirb) [inline avaframe/com1DFA/com1DFA.py:1816]: sorted(inDir.glob('*.p')) do you need a list here? sorted(list(inDir.glob('*.p')))

File: avaframe/com1DFA/com1DFA.py [hunk 2]
```diff
@@ -1916,17 +1916,17 @@ def exportFields(cfg, Tsave, fieldsList, demOri, outDir, logName):
                 resField = resField * 0.001
             dataName = logName + '_' + resType + '_' + 't%.2f' % (Tsave[countTime]) + '.asc'
             # create directory
-            outDirPeak = os.path.join(outDir, 'peakFiles', 'timeSteps')
+            outDirPeak = pathlib.Path(outDir, 'peakFiles', 'timeSteps')
             fU.makeADir(outDirPeak)
-            outFile = os.path.join(outDirPeak, dataName)
+            outFile = pathlib.Path(outDirPeak, dataName)
```
Comment (awirb) [inline avaframe/com1DFA/com1DFA.py:1921]: outFile = outDirPeak / dataName

File: avaframe/com1DFA/com1DFA.py [hunk 3]
```diff
@@ -1916,17 +1916,17 @@ def exportFields(cfg, Tsave, fieldsList, demOri, outDir, logName):
...
             IOf.writeResultToAsc(demOri['header'], resField, outFile, flip=True)
             if countTime == numberTimes:
                 log.debug('Results parameter: %s has been exported to Outputs/peakFiles for time step: %.2f - FINAL time step ' % (resType, Tsave[countTime]))
                 dataName = logName + '_' + resType + '.asc'
                 # create directory
-                outDirPeakAll = os.path.join(outDir, 'peakFiles')
+                outDirPeakAll = pathlib.Path(outDir, 'peakFiles')
                 fU.makeADir(outDirPeakAll)
-                outFile = os.path.join(outDirPeakAll, dataName)
+                outFile = pathlib.Path(outDirPeakAll, dataName)
```
Comment (awirb) [inline avaframe/com1DFA/com1DFA.py:1929]: outFile = outDirPeakAll / dataName

File: avaframe/in1Data/getInput.py [hunk 4]
```diff
@@ -168,33 +169,36 @@ def getInputDataCom1DFA(avaDir, cfg):
...
+    inputDir = pathlib.Path(avaDir, 'Inputs')
 
     # Set flag if there is an entrainment or resistance area
     entResInfo= {'flagEnt': 'No', 'flagRes': 'No'}
 
     # Initialise release areas, default is to look for shapefiles
     if cfg['flagDev'] == 'True':
         releaseDir = 'devREL'
-        relFiles = glob.glob(inputDir+os.sep + releaseDir+os.sep + '*.shp')
+        releaseDir = pathlib.Path(inputDir, 'devREL')
```
Comment (awirb) [inline avaframe/in1Data/getInput.py:180]: inputDir already a pathlib path so releaseDir = inputDir / 'devREL' enough

File: avaframe/tests/test_ana3AIMEC.py [hunk 5]
```diff
@@ -1,29 +1,43 @@
...
-    dirname = os.path.dirname(__file__)
-    dataRef = os.path.join(dirname, 'data', 'refTestAimecTopo.asc')
-    dataSim = os.path.join(dirname, 'data', 'simTestAimecTopo.asc')
-    dataMass = os.path.join(dirname, 'data', '000000.txt')
-    dataMass1 = os.path.join(dirname, 'data', '000001.txt')
+    dirname = pathlib.Path(__file__).parents[0]
+    dataRef = pathlib.Path(dirname, 'data', 'refTestAimecTopo.asc')
+    dataSim = pathlib.Path(dirname, 'data', 'simTestAimecTopo.asc')
+    dataMass = pathlib.Path(dirname, 'data', '000000.txt')
+    dataMass1 = pathlib.Path(dirname, 'data', '000001.txt')
```
Comment (awirb) [inline avaframe/tests/test_ana3AIMEC.py:35]: same here already pathlib path so dirname / 'data' .... enough

**Review by awirb — CHANGES_REQUESTED** [review]
General comment: 
please go through and if the path in the pathlib.Path() is already a pathlib path object remove the pathlib.Path() and use newPath = path / 'data' / ... (I added this comment at several places but not all)

please check if you need path.stem or path.name when replacing os.path.basename as I think path.name is required instead of path.stem as this is then without the file extension

File: avaframe/runCom1DFA.py [hunk 6]
```diff
@@ -116,7 +117,7 @@ def runCom1DFA(avaDir='', cfgFile='', relThField='', variationDict=''):
             cfgUtils.writeCfgFile(avalancheDir, com1DFA, cfg, fileName='%s_butModified' % simHash)
 
     # Set directory for report
-    reportDir = os.path.join(avalancheDir, 'Outputs', 'com1DFA', 'reports')
+    reportDir = pathlib.Path(avalancheDir, 'Outputs', 'com1DFA', 'reports')
```
Comment (awirb) [inline avaframe/runCom1DFA.py:120]: should this be 'com1DFA' hardcoded or should we use modName?

File: avaframe/tests/test_ana3AIMEC.py [hunk 7]
```diff
@@ -79,38 +92,40 @@ def test_makeDomainTransfo(capfd):
...
 
     cfgPath['contCmap'] = True
 
-    pathResult = os.path.join(dirname, 'results')
+    pathResult = dirname / 'results'
     cfgPath['pathResult'] = pathResult
 
     cfgPath['projectName'] = 'testAna3Aimec'
-    pathName = os.path.basename(profileLayer[0])
+    pathName = profileLayer[0].stem
```
Comment (awirb) [inline avaframe/tests/test_ana3AIMEC.py:128]: name?

File: avaframe/tests/test_com2AB.py [hunk 8]
```diff
@@ -113,10 +113,10 @@ def test_com2ABMain(capfd):
     '''Simple test for function com2ABMain'''
     # load and prepare Inputs
     listNames = ['avaHockeySmall', 'avaHockeyChannel', 'avaBowl']
-    dirname = os.path.dirname(__file__)
+    dirname = pathlib.Path(__file__).parents[0]
     for name in listNames:
-        avalancheDir = os.path.join(dirname, '..', 'data', name)
-        saveOutPathRef = os.path.join(dirname, '..', '..', 'benchmarks', name + 'ABPytest')
+        avalancheDir = pathlib.Path(dirname, '..', 'data', name)
+        saveOutPathRef = pathlib.Path(dirname, '..', '..', 'benchmarks', name + 'ABPytest')
```
Comment (awirb) [inline avaframe/tests/test_com2AB.py:119]: dirName is already a pathlib object so you could use dirname / '..' / 'data' /name

File: avaframe/tests/test_dfa2Aimec.py [hunk 9]
```diff
@@ -61,7 +53,7 @@ def test_dfaComp2Aimec(tmp_path):
     pathDTest['ppr'] = [pathData / 'release1HS_ent_dfa_0.15500_ppr.asc', pathData2 / 'release1HS_ent_dfa_67dc2dc10a_ppr.asc']
     pathDTest['pfd'] = [pathData / 'release1HS_ent_dfa_0.15500_pfd.asc', pathData2 / 'release1HS_ent_dfa_67dc2dc10a_pfd.asc']
     pathDTest['pfv'] = [pathData / 'release1HS_ent_dfa_0.15500_pfv.asc', pathData2 / 'release1HS_ent_dfa_67dc2dc10a_pfv.asc']
-    pathDTest['massBal'] = [os.path.join(testPath, 'Outputs', 'com1DFAOrig', 'mass_release1HS_ent_dfa_0.15500.txt'), os.path.join(testPath, 'Outputs', 'com1DFA', 'mass_release1HS_ent_dfa_67dc2dc10a.txt')]
+    pathDTest['massBal'] = [pathlib.Path(testPath, 'Outputs', 'com1DFAOrig', 'mass_release1HS_ent_dfa_0.15500.txt'), pathlib.Path(testPath, 'Outputs', 'com1DFA', 'mass_release1HS_ent_dfa_67dc2dc10a.txt')]
```
Comment (awirb) [inline avaframe/tests/test_dfa2Aimec.py:56]: testPath already as pathlib object..

File: avaframe/tests/test_in1Data.py [hunk 10]
```diff
@@ -18,7 +19,7 @@ def test_getInputData(tmp_path):
     # get input data
     dirPath = os.path.dirname(__file__)
     avaName = 'avaHockeyChannel'
-    avaDir  = os.path.join(tmp_path, avaName)
+    avaDir = os.path.join(tmp_path, avaName)
```
Comment (awirb) [inline avaframe/tests/test_in1Data.py:22]: consider using pathlib?

File: avaframe/tests/test_shpConversion.py [hunk 11]
```diff
@@ -190,23 +190,23 @@ def test_removeFeature(capfd):
 
 def test_readLine(capfd):
     '''Simple test for function readLine'''
-    dirname = os.path.dirname(__file__)
-    demFileName = os.path.join(dirname, 'data', 'testShpConv', 'testShpConv.asc')
+    dirname = pathlib.Path(__file__).parents[0]
+    demFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testShpConv.asc')
     dem = ascUtils.readRaster(demFileName)
-    shpFileName = os.path.join(dirname, 'data', 'testShpConv', 'testLine.shp')
+    shpFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testLine.shp')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:196]: dirname already a pathlib object

File: avaframe/tests/test_shpConversion.py [hunk 12]
```diff
@@ -190,23 +190,23 @@ def test_removeFeature(capfd):
...
 
     # do we react properly when the input line exceeds the dem?
     with pytest.raises(Exception) as e:
         assert shpConv.readLine(shpFileName, '', dem)
     assert str(e.value) == "Nan Value encountered. Try with another path"
 
     # do we react properly when the input line exceeds the dem?
-    shpFileName = os.path.join(dirname, 'data', 'testShpConv', 'testLineOut.shp')
+    shpFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testLineOut.shp')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:204]: same here

File: avaframe/tests/test_shpConversion.py [hunk 13]
```diff
@@ -190,23 +190,23 @@ def test_removeFeature(capfd):
...
     with pytest.raises(Exception) as e:
         assert shpConv.readLine(shpFileName, '', dem)
     assert str(e.value) == "The avalanche path exceeds dem extent. Try with another path"
 
-    shpFileName = os.path.join(dirname, 'data', 'testShpConv', 'testLineGood.shp')
+    shpFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testLineGood.shp')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:209]: same here

File: avaframe/tests/test_shpConversion.py [hunk 14]
```diff
@@ -241,18 +241,18 @@ def test_readLine(capfd):
 
 def test_readPoints(capfd):
     '''Simple test for function readPoints'''
-    dirname = os.path.dirname(__file__)
-    demFileName = os.path.join(dirname, 'data', 'testShpConv', 'testShpConv.asc')
+    dirname = pathlib.Path(__file__).parents[0]
+    demFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testShpConv.asc')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:245]: same here

File: avaframe/tests/test_shpConversion.py [hunk 15]
```diff
@@ -241,18 +241,18 @@ def test_readLine(capfd):
...
     dem = ascUtils.readRaster(demFileName)
 
     # do we react properly when the input point exceeds the dem?
-    shpFileName = os.path.join(dirname, 'data', 'testShpConv', 'testLine.shp')
+    shpFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testLine.shp')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:249]: same here

File: avaframe/tests/test_shpConversion.py [hunk 16]
```diff
@@ -241,18 +241,18 @@ def test_readLine(capfd):
...
     with pytest.raises(Exception) as e:
         assert shpConv.readPoints(shpFileName, dem)
     assert str(e.value) == 'Nan Value encountered. Try with another split point'
 
     # do we react properly when the input point exceeds the dem?
-    shpFileName = os.path.join(dirname, 'data', 'testShpConv', 'testLineOut.shp')
+    shpFileName = pathlib.Path(dirname, 'data', 'testShpConv', 'testLineOut.shp')
```
Comment (awirb) [inline avaframe/tests/test_shpConversion.py:255]: same here

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/tests/test_in1Data.py (hunk 10 above)
Comment (matthiasto) [reply avaframe/tests/test_in1Data.py:22]: getInputData is for com1DFAOrig which still works with os... so I did not change it

**Pull request merged**